### PR TITLE
Chunk refactor TTL

### PR DESCRIPTION
**Pull request by janos**
Fixes for the Chunk refactor.

### CODE REVIEW

File: swarm/network/fetcher.go [hunk 1]
```diff
@@ -49,7 +52,29 @@ type Request struct {
...
+	if !ok {
+		return false
+	}
+	t, ok := val.(time.Time)
+	if ok && time.Now().After(t.Add(RequestTimeout)) {
+		// deadine expired
+		r.peersToSkip.Delete(nodeID)
+		return false
+	}
+	return true
```
Comment (zelig) [inline swarm/network/fetcher.go:83]: lets create an issue to add unit tests for this

File: swarm/network/stream/syncer.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func (s *SwarmSyncerServer) SetNextBatch(from, to uint64) ([]byte, uint64, uint6
 		if (to < from && to != 0) || from > s.sessionAt {
 			return nil, 0, 0, nil, nil
 		}
-		if to > s.sessionAt {
+		if to == 0 || to > s.sessionAt {
```
Comment (zelig) [inline swarm/network/stream/syncer.go:106]: create an issue to simplify nextBatch functions both client and server

**Review by zelig — APPROVED** [review]

File: swarm/network/fetcher.go [hunk 3]
```diff
@@ -49,7 +52,29 @@ type Request struct {
...
+
+func NewRequest(addr storage.Address, skipCheck bool, peersToSkip *sync.Map) *Request {
+	return &Request{
+		Addr:        addr,
+		SkipCheck:   skipCheck,
+		peersToSkip: peersToSkip,
+	}
+}
+
+func (r *Request) SkipPeer(nodeID string) bool {
```
Comment (holisticode) [inline swarm/network/fetcher.go]: Please add a short comment about this function - its purpose and what it does

File: swarm/network/fetcher.go [hunk 4]
```diff
@@ -184,7 +209,7 @@ func (f *Fetcher) run(ctx context.Context, peers *sync.Map) {
 			var err error
 			sources, err = f.doRequest(ctx, gone, peers, sources)
 			if err != nil {
-				log.Debug("unable to request", "request addr", f.addr, "err", err)
+				log.Warn("unable to request", "request addr", f.addr, "err", err)
```
Comment (holisticode) [inline swarm/network/fetcher.go:218]: I wonder, should we do something about it at this point? Is having a `Warn` enough? I am aware that we may not be able to do much, I guess `panic` and stop the node is too radical.

**Review by holisticode — APPROVED** [review]
LGTM, please just address the 2 comments I left

File: swarm/network/fetcher.go (hunk 4 above)
Comment (zelig) [reply swarm/network/fetcher.go:218]: it is totally acceptable to have this when we are not connected

File: swarm/network/fetcher.go (hunk 1 above)
Comment (janos) [reply swarm/network/fetcher.go:83]: Created https://github.com/ethersphere/go-ethereum/issues/921.

File: swarm/network/stream/syncer.go (hunk 2 above)
Comment (janos) [reply swarm/network/stream/syncer.go:106]: Created https://github.com/ethersphere/go-ethereum/issues/922.

**Review by zelig — APPROVED** [review]

**Pull request merged**